src/core/src/log.cpp

```diff
@@ -0,0 +1,38 @@
+// Copyright (C) 2018-2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/core/log.hpp"
+
+#include "openvino/core/log_util.hpp"
+
+namespace ov::util {
+namespace {
+LogCallback default_callback{[](std::string_view s) {
+    std::cout << s << std::endl;
+}};
+
+LogCallback silent_callback{[](std::string_view s) {}};
+
+LogCallback* current_callback = &default_callback;
+}  // namespace
+
+OPENVINO_API
+LogCallback& get_log_callback() {
```

Comment:
Return const ref? 
If non const object returned here is any risk that user e.g. modify default callback (or any other) which is internal.